### ISSUE TITLE
Provide an option to control space between line number and context content (respect `statuscolumn`)

### ISSUE THREAD

**Issue by registerGen** [nvim-treesitter/nvim-treesitter-context#211]
With the new `statuscolumn` feature, we are able to customize the space between the line number and the code.

But unfortunately, in this plugin the space between line number and context content is fixed, shown as below.

![image](https://user-images.githubusercontent.com/62944333/223747115-7fa71b48-ef33-4876-afc4-e1f55df34762.png)

It would be great if we can control this so that line numbers are aligned.

**Comment by registerGen** [comment]
Oh, should this issue be titled as "respect `statuscolumn`", which is more general?

**Comment by registerGen** [comment]
Any ideas on this?

**Comment by lewis6991** [comment]
Nope. This isn't something I have much interest to support. Statuscolumn is pretty complicated.

**Comment by registerGen** [comment]
OK. Thanks for the explanation.

**Comment by akinsho** [comment]
I have a similar-ish issue which I thought might relate to the OP's issue or help resolve it if it's similar enough to mine in that I'm also using the statucolumn feature, but have found that the offset of the context menu when `line_numbers = true` can be "incorrect" if the position of the numbers when using a custom statuscolumn can different to where the statuscolumn ends

<img width="487" alt="image" src="https://user-images.githubusercontent.com/22454918/228982335-34e71c43-0878-48b5-b1ab-0aca78f5c4ca.png">

In the screenshot above I have about 3 columns to the right of the statuscolumn which are intended for signs/folds. So whilst the `textoff` value which is being used is technically correct it doesn't return what column the numbers end at.

So perhaps rather than needing/wanting to actually render the exact same statuscolumn since I think the the bells and whistles of that are (IMO) not needed in the context menu, perhaps the user could just be able to use the `line_number` option to specify the width of the gutter themselves.

TLDR: *I think maybe the issue here is just about alignment* and an option could be extended or added to allow the user to specify what the `gutter_width` is themselves. Maybe rather than the default function a user could pass in their own.

e.g.
```lua
config = {
  gutter_width = 5
  -- or as a function
  gutter_width = function()
	return calculate_nr_column_from_status_col()
  end
}
```
